src/applications/third-party-apps/actions/index.js

```diff
@@ -0,0 +1,88 @@
+// Node modules.
+// import recordEvent from 'platform/monitoring/record-event';
+// Relative imports.
+import { fetchResultsApi } from '../api';
+import { updateQueryParams } from '../helpers';
+import {
+  FETCH_RESULTS,
+  FETCH_RESULTS_FAILURE,
+  FETCH_RESULTS_SUCCESS,
+} from '../constants';
+
+// ============
+// Fetch Results (via API)
+// ============
+export const fetchResultsAction = (options = {}) => ({
+  options,
+  type: FETCH_RESULTS,
+});
+
+export const fetchResultsFailure = error => ({
+  error,
+  type: FETCH_RESULTS_FAILURE,
+});
+
+export const fetchResultsSuccess = response => ({
+  response,
+  type: FETCH_RESULTS_SUCCESS,
+});
+
+// ============
+// Redux Thunks
+// ============
+export const fetchResultsThunk = (options = {}) => async dispatch => {
+  // Derive options properties.
+  const category = options?.category || null;
+  const platform = options?.platform || null;
+  const hideFetchingState = options?.hideFetchingState;
+  const history = options?.history;
+  const location = options?.location;
+  const page = options?.page || 1;
+  const perPage = options?.perPage || 10;
+  // const trackSearch = options?.trackSearch || false;
+
+  const queryParamsLookup = { category, platform };
+
+  // Change the `fetching` state in our store.
+  dispatch(
+    fetchResultsAction({
+      ...queryParamsLookup,
+      hideFetchingState,
+      page,
+    }),
+  );
+
+  // Update query params.
+  updateQueryParams(history, location, queryParamsLookup);
+
+  try {
+    // Attempt to make the API request to retreive results.
+    const response = await fetchResultsApi({
+      category,
+      platform,
+      page,
+      perPage,
+      mockRequest: true,
```

Comment:
Need to remove this once we're not mocking the request anymore and we integrate with the new Lighthouse API endpoint